### PR TITLE
Use discord checkboxes, add links to plugin description, allow non-merging imports

### PR DESCRIPTION
**Pull request by skykittenpuppy**
yippee

still probably needs a way to preview each step of the regex (seperated applyRules into applyRule to allow for that down the line)
and then- the single row isn't the best for usability

### CODE REVIEW

**Comment by skykittenpuppy** [comment]
it looks so pretty- it looks so functional- but its so ugly code wise i am so sorry autumn please save me

File: src/plugins/textReplace/index.tsx [hunk 1]
```diff
@@ -333,8 +415,30 @@ const TEXT_REPLACE_RULES_CHANNEL_ID = "1102784112584040479";
 
 export default definePlugin({
     name: "TextReplace",
-    description: "Replace text in your messages. You can find pre-made rules in the #textreplace-rules channel in Vencord's Server",
-    authors: [Devs.AutumnVN, Devs.TheKodeToad],
+    description: "Replace text in your messages.",
+    authors: [Devs.TheKodeToad, Devs.skykittenpuppy, Devs.AutumnVN],
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:419]:
```suggestion
    authors: [Devs.AutumnVN, Devs.TheKodeToad, Devs.skykittenpuppy],
```
dont change authors order

File: src/plugins/textReplace/index.tsx [hunk 2]
```diff
@@ -277,53 +363,49 @@ async function tryImport(str: string, update: () => void) {
             if (typeof rule.onlyIfIncludes !== "string") throw new Error("A rule is missing onlyIfIncludes.");
             if (typeof rule.isRegex !== "boolean") throw new Error("A rule is missing isRegex.");
             textReplaceRules.push(rule);
+            if (!merge) { // Wow this is gross! sucks that it works
+                merge = true;
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:367]: what, why change it to true if it false

File: src/plugins/textReplace/index.tsx [hunk 3]
```diff
@@ -277,53 +363,49 @@ async function tryImport(str: string, update: () => void) {
...
+                textReplaceRules = [rule];
+            }
             await DataStore.set(TEXT_REPLACE_KEY, textReplaceRules);
             update();
         }
         Toasts.show({
             type: Toasts.Type.SUCCESS,
-            message: "Successfully imported & merged text replace rules.",
+            message: "Successfully " + merge ? "merged" : "imported" + " text replace rules.",
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:375]: somehow it always show one single word "merged"

File: src/plugins/textReplace/index.tsx [hunk 4]
```diff
@@ -277,53 +363,49 @@ async function tryImport(str: string, update: () => void) {
...
-    return (
-        <>
-            <Forms.FormTitle tag="h4">Test Rules</Forms.FormTitle>
-            <TextInput placeholder="Type a message" onChange={setValue} />
-            <TextInput placeholder="Message with rules applied" editable={false} value={applyRules(value)} />
-        </>
-    );
+function applyRule(rule: Rule, content: string): string {
+    if (!rule.isEnabled || !rule.find) return content;
+    if (rule.isRegex && !rule.onlyIfIncludes || /* MAKE THIS A REGEX CHECK */ content.includes(rule.onlyIfIncludes) /* REGEX CHECK */) {
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:390]: .

File: src/plugins/textReplace/index.tsx [hunk 5]
```diff
@@ -140,55 +207,141 @@ function TextReplace({ update }: { update: () => void; }) {
...
+                    <Button
+                        onClick={exportRules}
+                    >
+                        Export Rules
+                    </Button>
+                    <Button
+                        onClick={newRule}
+                        color={Button.Colors.GREEN}
+                    >
+                        New Rule
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:248]:
```suggestion
                        Create New Rule
```
i want it longer than others

File: src/plugins/textReplace/index.tsx [hunk 6]
```diff
@@ -112,12 +115,76 @@ function Input({ initialValue, onChange, placeholder }: {
...
 
         await DataStore.set(TEXT_REPLACE_KEY, textReplaceRules);
         update();
     }
+    async function onClickMoveUp(index: number) {
+        if (index === 0) return;
+        [textReplaceRules[index], textReplaceRules[index - 1]] = [textReplaceRules[index - 1], textReplaceRules[index]];
+
+        await DataStore.set(TEXT_REPLACE_KEY, textReplaceRules);
+        update(); // BIG PROBLEM: doesn't update the textboxes, but it does work
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:179]: .

File: src/plugins/textReplace/index.tsx [hunk 7]
```diff
@@ -140,55 +207,141 @@ function TextReplace({ update }: { update: () => void; }) {
...
+    }, [value, textReplaceRules]);
+
     return (
         <>
-            <Forms.FormTitle tag="h5" style={{ marginBottom: 0 }}>IS ENABLED</Forms.FormTitle >
-            <Forms.FormTitle tag="h5" style={{ position: "absolute", right: "30px" }}>IS REGEX</Forms.FormTitle >
+            <Flex style={{ justifyContent: "space-between" }}>
+                <Flex>
+                    <Button
+                        onClick={importRules}
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:230]:
```suggestion
                        onClick={() => rulesFromFile(false)}
```

File: src/plugins/textReplace/index.tsx [hunk 8]
```diff
@@ -140,55 +207,141 @@ function TextReplace({ update }: { update: () => void; }) {
...
+                    >
+                        Import Rules
+                    </Button>
+                    <Button
+                        onClick={mergeRules}
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:235]:
```suggestion
                        onClick={() => rulesFromFile(true)}
```

File: src/plugins/textReplace/index.tsx [hunk 9]
```diff
@@ -140,55 +207,141 @@ function TextReplace({ update }: { update: () => void; }) {
         update();
     }
 
+    const [value, setValue] = useState("");
+
+    const previews = useMemo(() => {
```
Comment (AutumnVN) [inline src/plugins/textReplace/index.tsx:212]: preview doesnt seem to create more when create new rule, import or merged

**Review by AutumnVN — COMMENTED** [review]
lol this pr changes more than my original pr

**Pull request merged**